### ISSUE TITLE
[BUG]Unable to deploy AKS with Proxy feature using Terraform

### ISSUE THREAD

**Issue by bkuliqi** [Azure/AKS#3044]
**Describe the bug**
Issue: Terraform throws an error when deploying AKS with Proxy configuration

Error: creating Cluster: (Managed Cluster Name "" / Resource Group ""): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="HTTPProxyTrustedCAPEMDecodeFailure" Message="failed to decode HTTPProxyConfig.TrustedCA to PEM after base64 decoding."
│
│ with azurerm_kubernetes_cluster.example-aks,
│ on main.tf line 22, in resource "azurerm_kubernetes_cluster" "example-aks":
│ 22: resource "azurerm_kubernetes_cluster" "example-aks" {

In our configuration we do not require to provide a cert as use http proxy, so we are setting trusted_ca to null or just removing trusted_ca altogether from the tf module. The same error occurs when setting trusted_ca = null or just omitting trusted_ca altogether. The terraform documentation mentions trusted_ca is optional:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster

Here is my AKS TF code for proxy settings:

http_proxy_config{
http_proxy ="http://myproxy.server.com:8080/"
https_proxy ="http://myproxy.server.com:8080/"
no_proxy = ["localhost","127.0.0.1"]
trusted_ca = null

}

It appears based on the error that either Terraform or Azure resource manager is thinking there is a cert provided and it cant decode, when in fact we are not providing a certificate.

**Comment by MrSimonEmms** [comment]
I'm getting the same issue as well. According to the [AKS documentation](https://docs.microsoft.com/en-us/azure/aks/http-proxy#configuring-an-http-proxy-using-azure-cli), this also says it's optional.

If, like me, you're not using HTTPS or using one that's publicly signed, you could get around this by generating one. I'm using this `ca.pem`:

```
LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJjRENDQVJXZ0F3SUJBZ0lSQU9LaG5Ba1lEaEJPTndvWlhXL2REam93Q2dZSUtvWkl6ajBFQXdJd0Z6RVYKTUJNR0ExVUVBeE1NWTJFdGFYTnpkV1Z5TFdOaE1CNFhEVEl5TURnd01URTFOVGd3TkZvWERUSXlNVEF6TURFMQpOVGd3TkZvd0Z6RVZNQk1HQTFVRUF4TU1ZMkV0YVhOemRXVnlMV05oTUZrd0V3WUhLb1pJemowQ0FRWUlLb1pJCnpqMERBUWNEUWdBRTZxU09xeXJ1VENrd21wQW1TbG9nVVFmQ1I3dVJWcC9VRmRGVzhmWE5mcFpXalA5K21ybGsKRmVOdGFPMkxNL3A3K2J0QmhXVHljMXFrOW9HbG9HcnFtYU5DTUVBd0RnWURWUjBQQVFIL0JBUURBZ0trTUE4RwpBMVVkRXdFQi93UUZNQU1CQWY4d0hRWURWUjBPQkJZRUZQM3NYS1MxWmlvWWMyVGRQdmh1N04vZzYxa21NQW9HCkNDcUdTTTQ5QkFNQ0Ewa0FNRVlDSVFEQnhyRTdPSmJxR3BzSyt4VEtLTENydjNieVhsK2dnaFM0NTUwQStvY24KZndJaEFNV1lBOFlvSHgwb1JmV3ZhcUg5T09kOXlwZHlRYnAwTkFwZ3M3bTZCekVzCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
```

**Comment by MrSimonEmms** [comment]
@bkuliqi what version of TF Azure and Azure CLI are you using? After much shenanigans, I've upgraded to `3.16.0` for the TF provider and `2.39.0` for Azure CLI and no longer get the issues you raised (I get another one which I don't think it relevant to this issue)

**Comment by bkuliqi** [comment]
thanks @MrSimonEmms, we were using v3.12, latest at the time. This got fixed with v3.13 and fix works. 
https://github.com/hashicorp/terraform-provider-azurerm/issues/17442#event-7105969029
